client/components/room/Video.jsx

```diff
@@ -301,8 +307,67 @@ class Video extends React.Component {
     }
   }
 
-  // Added 'muted' here to help reduce audio feedback but not entirely sure it will work
+  requestTurn() {
+    const turnURL = 'https://numb.viagenie.ca?username=andy.yeo@gmail.com&key=hackreactor';
+    let turnExists = false;
+    for (let i in configuration.iceServers) {
+      if (configuration.iceServers[i].urls.substr(0, 5) === 'turn:') {
+        turnExists = true;
+        turnReady = true;
+        break;
+      }
+    }
+    if (!turnExists) {
+      console.log('Getting TURN server from ', turnURL);
+      const xhr = new XMLHttpRequest();
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4 && xhr.status === 200) {
+          var turnServer = JSON.parse(xhr.responseText);
+          console.log('Got TURN server: ', turnServer);
+          configuration.iceServers.push({
+            'url': 'turn:' + turnServer.username + '@' + turnServer.turn,
+            'credential': turnServer.password
+          });
+          turnReady = true;
+        }
+      };
+      xhr.open('GET', turnURL, true);
+      xhr.send();
+    }
+  }
+
+  copyTextToClipboard(text) {
+    const textArea = document.createElement('textarea');
+    textArea.value = text;
+    document.body.appendChild(textArea);
+    textArea.select();
+
+    try {
+      const successful = document.execCommand('copy');
+      const msg = successful ? 'successful' : 'unsuccessful';
+      console.log(`Copying text command was ${msg}`);
+    } catch (err) {
+      console.log('Oops, unable to copy');
+    }
+    document.body.removeChild(textArea);
+  }
+
+  handleCopy() {
+    document.querySelector('.js-textareacopybtn').addEventListener('click', (event) => {
+      this.copyTextToClipboard(window.location.href);
+    });
+  }
+
   render() {
+    let sharing = null;
```

Comment:
Only feedback I have is maybe converting this to a method of the class, rather than a function within the render statement, if only to keep the render function itself relatively clean. Just depends on personal preference really though.